go-controller/pkg/libovsdbops/datapath_binding.go

```diff
@@ -0,0 +1,31 @@
+package libovsdbops
+
+import (
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/sbdb"
+)
+
+type datapathBindingPredicate func(*sbdb.DatapathBinding) bool
+
+// GetDatapathBindingWithPredicate looks up a datapath binding from the cache
+// based on a given predicate. If none or multiple are found, an error is
+// returned.
+func GetDatapathBindingWithPredicate(sbClient libovsdbclient.Client, p datapathBindingPredicate) (*sbdb.DatapathBinding, error) {
+	found := []*sbdb.DatapathBinding{}
+	opModel := OperationModel{
+		ModelPredicate: p,
+		ExistingResult: &found,
+		OnModelUpdates: nil, // no update
+		ErrNotFound:    true,
+		BulkOp:         false,
+	}
+
+	m := NewModelClient(sbClient)
+	_, err := m.CreateOrUpdate(opModel)
+	if err != nil {
+		return nil, err
+	}
+
+	return found[0], nil
```

Comment:
a) if `ErrNotFound: true` we are guaranteed to have `len(found)>0` otherwise a `notFound` error should be returned. I prefer to keep it as is if you dont't have a strong opinion about it.
b) `BulkOp: false` guarantees  `len(found)<2`